sites/authentication/includes/authentication.html

```diff
@@ -0,0 +1,33 @@
+<section  id="wb-so" class="pull-right mrgn-rght-md btn-signin">
+	<h2 class="wb-inv">
+	{% unless page.contextSignin %}
+		{{ i18nText-signOnOff }}
+	{% else %}
+		{{ i18nText-signIn }}
+	{% endunless %}
+	</h2>
+{%- if page.contextSignin -%}
+	<a class="btn btn-primary" href="{{ page.contextSignin.link }}">
+	{%- if page.contextSignin.label and page.contextSignin.labelSm -%}
+			<span class="visible-xs">{{ page.contextSignin.label | truncate: 12, "" }}</span>
+			<span class="hidden-xs">{{ page.contextSignin.labelSm | truncate: 25, "" }}</span>
+	{%- elsif page.contextSignin.label == nil and page.contextSignin.labelSm -%}
+			<span class="visible-xs">{{ i18nText-signIn }}</span>
+			<span class="hidden-xs">{{ page.contextSignin.labelSm | truncate: 25, "" }}</span>
+	{%- elsif page.contextSignin.label and page.contextSignin.labelSm == nil -%}
+		{{ page.contextSignin.label | truncate: 12, "" }}
```

Comment:
It will simply the logic if before you create 2 "%assign%" variable for the labels. You can achieve that by using the capture tag https://shopify.github.io/liquid/tags/variable/#capture